docs/src/api/catalyst_api.md

```diff
@@ -178,6 +178,9 @@ substoichmat
 prodstoichmat
 netstoichmat
 reactionrates
+get_metadata_dict
```

Comment:
This seems like a different approach than what we use for symbolics. There we define custom get/has functions for each possible metadata type, not generic functions that take a symbol indicating the metadata we want. Why change the interface here?